src/gui/torrentcontentmodel.cpp

```diff
@@ -88,6 +96,40 @@ namespace
             return QIcon(iconPixmap);
         }
     };
+#elif defined(Q_OS_MAC)
+    // There is a similar bug on macOS, to be reported to Qt
+    // https://github.com/qbittorrent/qBittorrent/pull/6156#issuecomment-316302615
+    class MacFileIconProvider: public UnifiedFileIconProvider
+    {
+    public:
+        using QFileIconProvider::icon;
+        QIcon icon(const QFileInfo &info) const override
+        {
+            QSize sz(32, 32);
+            QPixmap pixmap = pixmapForExtension(info.suffix(), sz);
```

Comment:
Well, in my opinion it should be up to the OS to decide what to pass for a an empty extension… but since you have a default file icon specified somewhere, it is probably reasonable to add a check.

.tar.gz may give a <slight> difference, and I wanted to use [completeSuffix](http://doc.qt.io/qt-5/qfileinfo.html#completeSuffix) initially, but it reads the extension as the first dot in a file. In most cases it will be useless, and will simply waste time by passing invalid strings. Not worth adding it, in my opinion.

As for directories, there are few that need a special icon, e.g. various cfbundles like apps, kexts, frameworks, so on. They would unlikely reside in a torrent, so I don't think there is an issue in not colouring those.